assets/src/domain/shared/services/utils/switchTenseForDate/index.ts

```diff
@@ -0,0 +1,44 @@
+import { isNil } from 'ramda';
+import { parseISO } from 'date-fns';
+
+import { diff } from '@appServices/utilities';
+import { now } from '@sharedServices/utils/dateAndTime';
+
+interface CalendarDateLabels {
+	headerPast: string;
+	headerFuture: string;
+	footerPast: string;
+	footerFuture: string;
+}
+
+interface SwitchTenseForDate extends CalendarDateLabels {
+	endDate: string; // ISO string
+	startDate: string; // ISO string
+}
+
+const switchTenseForDate = ({
+	endDate,
+	footerFuture,
+	footerPast,
+	headerFuture,
+	headerPast,
+	startDate,
+}: SwitchTenseForDate): { footerText: string; headerText: string } => {
+	if (
+		isNil(endDate) ||
+		isNil(footerFuture) ||
+		isNil(footerPast) ||
+		isNil(headerFuture) ||
+		isNil(headerPast) ||
+		isNil(startDate)
+	) {
+		return { footerText: null, headerText: null };
+	}
```

Comment:
To avoid confusion or any edge case situations, I've checked each prop, hope that makes sense, if not lemme know.